File: arrays_and_hashing/longest_consecutive_sequence.py
# Q:128 - https://leetcode.com/problems/longest-consecutive-sequence/
from typing import List

class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        map, longest = {}, 0
        for num in nums:
            map[num] = True

        for num in nums:
            if (num - 1) in map:
                continue
            else:
                start = num
                currentLongest = 0
                while start in map:
                    currentLongest += 1
                    start += 1
                
                longest = max(longest, currentLongest)
        
        return longest


c = Solution()

nums1 = [100,4,200,1,3,2]
nums2 = [0,3,7,2,5,8,4,6,0,1]

print(c.longestConsecutive(nums1))
print(c.longestConsecutive(nums2))

File: arrays_and_hashing/longest_consecutive_sequence.js
// Q:128 - https://leetcode.com/problems/longest-consecutive-sequence/

/**
 * @param {number[]} nums
 * @return {number}
 */
var longestConsecutive = function (nums) {
  const map = new Map();
  let longestSequence = 0;

  for (let i = 0; i < nums.length; i++) {
    map.set(nums[i], true);
  }

  for (let num of nums) {
    if (map.has(num - 1)) {
      continue;
    } else {
      let startNum = num,
        currentSequence = 0;

      while (map.has(startNum)) {
        currentSequence += 1;
        startNum += 1;
      }

      longestSequence = Math.max(currentSequence, longestSequence);
    }
  }

  return longestSequence;
};
